src/main/scala/com/redis/RedisClient.scala

```diff
@@ -13,6 +15,16 @@ object RedisClient {
   case object SUM extends Aggregate
   case object MIN extends Aggregate
   case object MAX extends Aggregate
+
+  private def extractDatabaseNumber(connectionUri: java.net.URI) = {
+    Option(connectionUri.getPath).map(path =>
+      if (path.isEmpty) 0
+      else Try(Integer.parseInt(path.tail)) match {
+        case Success(dbNumber) => dbNumber
+        case Failure(e) => throw new IllegalArgumentException("Invalid database number", e)
+      })
+      .getOrElse(0)
+  }
 }
```

Comment:
Thanks for the feedback! :)
The `Option` handles the case when `connectionUri.getPath` is `null`, which would have to be added to your pattern match. So I think there isn't much to gain from removing it.
The wrapping `Try` is only to give a better error message, and could be removed to only propagate the `NumberFormatException` as-is. Should I simiplify that?